met-web/src/components/common/Modals/DayCalculator/index.tsx

```diff
@@ -0,0 +1,703 @@
+import React, { useState } from 'react';
+import { Modal, Grid, Stack, IconButton, useMediaQuery, Divider, TextField, Autocomplete } from '@mui/material';
+import { modalStyle, MetHeader1, MetHeader3, PrimaryButton } from 'components/common';
+import HighlightOffIcon from '@mui/icons-material/HighlightOff';
+import { When } from 'react-if';
+import dayjs from 'dayjs';
+import { muitheme, options } from './constants';
+import DayZeroRules from './dayZeroRules';
+
+interface DayCalcModalProps {
+    open: boolean;
+    updateModal: (open: boolean) => void;
+}
+interface ISelectOptions {
+    id: number;
+    label: string;
+}
+
+const DayCalculatorModal = ({ open, updateModal }: DayCalcModalProps) => {
+    const isSmallScreen = useMediaQuery(muitheme.breakpoints.down('lg'));
+
+    // variable to capture the option selected within Calculation Type
+    const [selectedOption, setSelectedOption] = useState<ISelectOptions | null>(options[0]);
+
+    // variable to throw error in case suspension or resumption dates are blank
+    const [dayCalcError, setdayCalcError] = useState({
+        suspensionDate: false,
+        resumptionDate: false,
+    });
+    const validateDayCalcInputs = () => {
+        const errors = {
+            suspensionDate: !suspensionDate,
+            resumptionDate: !resumptionDate,
+        };
+        setdayCalcError(errors);
+
+        return Object.values(errors).some((isError: unknown) => isError);
+    };
+
+    // variable to show or hide the day zero rules
+    const [showHideStatus, setShowHideStatus] = useState(false);
+
+    // variable to capture changes on the input fields
+    const [dayCalcData, setDayCalcData] = useState({
+        startDate: '',
+        endDate: '',
+        suspensionDate: '',
+        resumptionDate: '',
+        numberOfDays: 0,
+    });
+    const { startDate, endDate, suspensionDate, resumptionDate, numberOfDays } = dayCalcData;
+    const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+        setDayCalcData({
+            ...dayCalcData,
+            [e.target.name]: e.target.value,
+        });
+        setdayCalcError({
+            ...dayCalcError,
+            [e.target.name]: false,
+        });
+    };
+
+    // reset to default
+    const resetToDefault = () => {
+        setDayCalcData({
+            startDate: '',
+            endDate: '',
+            suspensionDate: '',
+            resumptionDate: '',
+            numberOfDays: 0,
+        });
+    };
+
+    // calculator
+    const calculator = () => {
+        let calcStartDate = startDate;
+        let calcEndDate = endDate;
+        let calcNumberOfDays = numberOfDays;
+        if (selectedOption?.label == options[0].label) {
+            if (startDate && endDate && !numberOfDays) {
```

Comment:
Can we try breaking every logic block into separate function with meaningful name like https://github.com/ryanmcdermott/clean-code-javascript#functions-should-do-one-thing

Reply:
you can try this as well https://github.com/ryanmcdermott/clean-code-javascript#encapsulate-conditionals

Reply:
updated as per review comments